### PR TITLE
Format MaxNLocator with numpydoc

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1833,32 +1833,30 @@ class MaxNLocator(Locator):
 
     def __init__(self, *args, **kwargs):
         """
-        Keyword args:
-
-        *nbins*
+        Parameters
+        ----------
+        nbins : int or 'auto', optional, default=10
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: default: 10

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -1833,32 +1833,30 @@ class MaxNLocator(Locator):
...
             Maximum number of intervals; one less than max number of
             ticks.  If the string `'auto'`, the number of bins will be
             automatically determined based on the length of the axis.
 
-        *steps*
+        steps: array-like of float, optional
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: probably "sequence of float" as it has to be 1D and can't be a scalar either.

Comment (timhoffm) [reply lib/matplotlib/ticker.py]: Changed to just `array-like`.

This is what we're recently using https://matplotlib.org/devdocs/devel/documenting_mpl.html#parameter-type-descriptions and is consistent with numpy and pandas.

Comment (anntzer) [reply lib/matplotlib/ticker.py]: Not really worth relitigating this here but I tend to think that "array-like" implies "any dimensionality" and that "list of ..." or "sequence of ..." is nicer than "1D array-like of ...".

Comment (timhoffm) [reply lib/matplotlib/ticker.py]: Just for reference, pandas and numpy us "array-like" in the 1D sense
https://pandas.pydata.org/pandas-docs/version/0.23.4/generated/pandas.Series.html
https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.interp.html

**Pull request merged**